tiledb/api/src/group/mod.rs

```diff
@@ -0,0 +1,556 @@
+use std::ops::Deref;
+
+use crate::config::{Config, RawConfig};
+use crate::context::ObjectType;
+use crate::{Context, ContextBound, Datatype};
+
+extern crate tiledb_sys as ffi;
+use crate::string::{RawTDBString, TDBString};
+use crate::Result as TileDBResult;
+pub type QueryType = crate::array::Mode;
+use crate::metadata::Metadata;
+
+pub struct GroupInfo {
+    pub uri: String,
+    pub group_type: ObjectType,
+    pub name: String,
+}
+
+pub(crate) struct RawGroup {
+    ffi: *mut ffi::tiledb_group_t,
+}
+
+impl RawGroup {
+    pub fn new(ffi: *mut ffi::tiledb_group_t) -> Self {
+        RawGroup { ffi }
+    }
+}
+
+impl Deref for RawGroup {
+    type Target = *mut ffi::tiledb_group_t;
+    fn deref(&self) -> &Self::Target {
+        &self.ffi
+    }
+}
+
+impl Drop for RawGroup {
+    fn drop(&mut self) {
+        unsafe { ffi::tiledb_group_free(&mut self.ffi) }
+    }
+}
+
+pub struct Group<'ctx> {
+    context: &'ctx Context,
+    raw: RawGroup,
+}
+
+impl<'ctx> ContextBound<'ctx> for Group<'ctx> {
```

Comment:
When you update your branch with `main` you can `#[derive(ContextBound)]` instead